components/elm/src/biogeophys/CanopyFluxesMod.F90

```diff
@@ -641,7 +646,7 @@ subroutine CanopyFluxes(bounds,  num_nolakeurbanp, filter_nolakeurbanp, &
                   ! Translate vol_liq_so and eff_porosity into h2osoi_liq_so and h2osoi_liq_sat and calculate deficit
                   h2osoi_liq_so  = vol_liq_so * denh2o * col_pp%dz(c,j)
                   h2osoi_liq_sat = eff_porosity(c,j) * denh2o * col_pp%dz(c,j)
-                  deficit        = max((h2osoi_liq_so + ldomain%firrig(g)*(h2osoi_liq_sat - h2osoi_liq_so)) - h2osoi_liq(c,j), 0._r8)
+                  deficit        = max((h2osoi_liq_so + firrig(g,tpu_ind)*(h2osoi_liq_sat - h2osoi_liq_so)) - h2osoi_liq(c,j), 0._r8)
```

Comment:
Will 'firrig' be the same in different topo-units for each grid? Same question to the other two variables 'f_grd' and 'f_surf'.

Reply:
@hydrotian 

> Is it BFB ?

@rljacob yes it BFB. I just run SMS test and all passed.

Reply:
@hydrotian the implementation of the topographic units has a flag to check whether the surface data has topounits or not. If the surface data file has no topounits the code assigns a single topounit per grid. So, the changes I have implemented for the firrig, f_surf, and f_grnd are able to read both surface data files.

Reply:
@hydrotian the implementation allows both options: topounits of the same grid sharing the same value and topounits of the same grid assigned different values.

Reply:
Thanks for the explainations.